quickwit/quickwit-storage/src/local_file_storage.rs

```diff
@@ -177,17 +172,64 @@ impl Storage for LocalFileStorage {
 
     async fn delete(&self, path: &Path) -> StorageResult<()> {
         let full_path = self.root.join(path);
-        missing_file_is_ok(fs::remove_file(full_path).await)?;
-        let parent = path.parent();
-        if parent.is_none() {
-            return Ok(());
-        }
-        if let Err(error) = delete_all_dirs(self.root.to_path_buf(), parent.unwrap()).await {
-            warn!(error = ?error, path = %path.display(), "Failed to clean up parent directories.");
+        ignore_error_kind!(ErrorKind::NotFound, fs::remove_file(full_path).await)?;
+        if let Some(parent) = path.parent() {
+            if let Err(error) = delete_all_dirs(&self.root, parent).await {
+                warn!(error=?error, path=%path.display(), "Failed to delete directory.");
+            }
         }
         Ok(())
     }
 
+    async fn bulk_delete<'a>(&self, paths: &[&'a Path]) -> Result<(), BulkDeleteError> {
```

Comment:
Proposition for doc
// Delete files defined by their `paths` by chunk of 10.
// If the directory that contains a file is empty, we delete the directory.